int32 port
int32 enable
int32 speed
int32 position